pkg/block/metrics.go

```diff
@@ -0,0 +1,133 @@
+package block
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/httputil"
+)
+
+type MetricsAdapter struct {
+	adapter Adapter
+}
+
+func NewMetricsAdapter(adapter Adapter) Adapter {
+	return &MetricsAdapter{adapter: adapter}
+}
+
+func (m *MetricsAdapter) InnerAdapter() Adapter {
+	return m.adapter
+}
+
+func (m *MetricsAdapter) Put(ctx context.Context, obj ObjectPointer, sizeBytes int64, reader io.Reader, opts PutOpts) error {
+	ctx = httputil.SetClientTrace(ctx, "block")
+	return m.adapter.Put(ctx, obj, sizeBytes, reader, opts)
+}
+
+func (m *MetricsAdapter) Get(ctx context.Context, obj ObjectPointer) (io.ReadCloser, error) {
+	ctx = httputil.SetClientTrace(ctx, "block")
+	return m.adapter.Get(ctx, obj)
+}
+
+func (m *MetricsAdapter) GetWalker(uri *url.URL) (Walker, error) {
+	return m.adapter.GetWalker(uri)
```

Comment:
I assume this does not add a client-trace field because this is currently never used on a server.  However this fragile: there is no way for future-me to know that it is incorrect to use Walk on the server side.
Please either mark on the context in advance which application it is on, or make sure that the client-trace field also works on lakectl.